### ISSUE TITLE
Failed to send shutdown command with windows 2012 on vmware-iso (0.12)

### ISSUE THREAD

**Issue by moodwrench** [mitchellh/packer#4191]
Version: Packer v0.11.0 and v0.12.0

```
# uname -a
Linux pack01d-ops-08.portal.webmd.com 2.6.32-504.30.3.el6.x86_64 #1 SMP Wed Jul 15 10:13:09 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
```

we've started seeing an issue that only seems to affect windows 2012 hosts using the vmware-iso builder on a remote ESXi machine.  I reported the issue on https://github.com/mitchellh/packer/issues/4171 but was told that this issue was fixed in 0.12.0.   Weve retested and the issue remains in 0.12.0

```
"type": "vmware-iso"
"remote_type": "esx5"
"shutdown_command": "shutdown /s /t 10 /f /d p:4:1 /c "Packer Shutdown"",
```

It seems that the build completes successfully with no communications issues or winrm issues, then times out connecting for the shutdown.

debug logs:

```
2016/11/15 14:19:18 ui: ==> vmware-iso: Gracefully halting virtual machine...
==> vmware-iso: Gracefully halting virtual machine...
2016/11/15 14:19:18 packer: 2016/11/15 14:19:18 Executing shutdown command: shutdown /s /t 10 /f /d p:4:1 /c "Packer Shutdown"
2016/11/15 14:20:21 ui error: ==> vmware-iso: Failed to send shutdown command: unknown error Post http://xxx.xxx.xxx.xxx:5985/wsman: dial tcp xxx.xxx.xxx.xxx:5985: getsockopt: connection timed out
2016/11/15 14:20:21 packer: 2016/11/15 14:20:21 opening new ssh session
==> vmware-iso: Failed to send shutdown command: unknown error Post http://xxx.xxx.xxx.xxx:5985/wsman: dial tcp xxx.xxx.xxx.xxx:5985: getsockopt: connection timed out
2016/11/15 14:20:21 packer: 2016/11/15 14:20:21 starting remote command: vim-cmd vmsvc/power.getstate 36
2016/11/15 14:20:22 ui: ==> vmware-iso: Destroying virtual machine...
==> vmware-iso: Destroying virtual machine...
2016/11/15 14:20:22 packer: 2016/11/15 14:20:22 opening new ssh session
2016/11/15 14:20:22 packer: 2016/11/15 14:20:22 starting remote command: vim-cmd vmsvc/destroy 36
2016/11/15 14:20:22 packer: 2016/11/15 14:20:22 opening new ssh session
2016/11/15 14:20:22 packer: 2016/11/15 14:20:22 starting remote command: test ! -e /vmfs/volumes/iso_datastore/windows_2012_r2-v1479232908
2016/11/15 14:20:22 packer: 2016/11/15 14:20:22 Remote command exited with '1': test ! -e /vmfs/volumes/iso_datastore/windows_2012_r2-v1479232908
2016/11/15 14:20:22 packer: 2016/11/15 14:20:22 opening new ssh session
2016/11/15 14:20:22 packer: 2016/11/15 14:20:22 starting remote command: test ! -e /vmfs/volumes/iso_datastore/windows_2012_r2-v1479232908
2016/11/15 14:20:22 packer: 2016/11/15 14:20:22 Remote command exited with '1': test ! -e /vmfs/volumes/iso_datastore/windows_2012_r2-v1479232908
2016/11/15 14:20:22 packer: 2016/11/15 14:20:22 opening new ssh session
2016/11/15 14:20:22 packer: 2016/11/15 14:20:22 starting remote command: test ! -e /vmfs/volumes/iso_datastore/windows_2012_r2-v1479232908
2016/11/15 14:20:22 packer: 2016/11/15 14:20:22 Deleting floppy disk: /tmp/packer222785091
2016/11/15 14:20:22 packer: 2016/11/15 14:20:22 opening new ssh session
2016/11/15 14:20:22 packer: 2016/11/15 14:20:22 starting remote command: test -e /vmfs/volumes/iso_datastore/windows_2012_r2-v1479232908
2016/11/15 14:20:23 packer: 2016/11/15 14:20:23 Remote command exited with '1': test -e /vmfs/volumes/iso_datastore/windows_2012_r2-v1479232908
Build 'vmware-iso' errored: Failed to send shutdown command: unknown error Post http://xxx.xxx.xxx.xxx:5985/wsman: dial tcp xxx.xxx.xxx.xxx:5985: getsockopt: connection timed out

==> Some builds didn't complete successfully and had errors:
2016/11/15 14:20:23 ui error: Build 'vmware-iso' errored: Failed to send shutdown command: unknown error Post http://xxx.xxx.xxx.xxx:5985/wsman: dial tcp xxx.xxx.xxx.xxx:5985: getsockopt: connection timed out
2016/11/15 14:20:23 Builds completed. Waiting on interrupt barrier...
2016/11/15 14:20:23 machine readable: error-count []string{"1"}
2016/11/15 14:20:23 ui error: 
==> Some builds didn't complete successfully and had errors:
--> vmware-iso: Failed to send shutdown command: unknown error Post http://xxx.xxx.xxx.xxx:5985/wsman: dial tcp xxx.xxx.xxx.xxx:5985: getsockopt: connection timed out

2016/11/15 14:20:23 machine readable: vmware-iso,error []string{"Failed to send shutdown command: unknown error Post http://xxx.xxx.xxx.xxx:5985/wsman: dial tcp xxx.xxx.xxx.xxx:5985: getsockopt: connection timed out"}
2016/11/15 14:20:23 ui error: --> vmware-iso: Failed to send shutdown command: unknown error Post http://xxx.xxx.xxx.xxx:5985/wsman: dial tcp xxx.xxx.xxx.xxx:5985: getsockopt: connection timed out
2016/11/15 14:20:23 ui: 
==> Builds finished but no artifacts were created.
```

**Comment by moodwrench** [comment]
updated output from 0.12:
2016/11/21 17:29:34 ui:     vmware-iso: SDelete is set for 1 pass.
    vmware-iso: SDelete is set for 1 pass.
2016/11/21 18:08:42 packer: 2016/11/21 18:08:42 [INFO] command 'set "CHEF_MSI_URL=" && set "CHEF_WIN_VER=" && set "PACKER_BUILDER_TYPE=vmware-iso" && set "PACKER_BUILD_NAME=vmware-iso" && set "RUBYINSTALL
ER_URL=" && set "UPDATE=true" && set "http_proxy=http://pack01d-ops-08.portal.webmd.com:3128" && set "https_proxy=https://pack01d-ops-08.portal.webmd.com:3128" && "c:/Windows/Temp/script.bat"' exited with
 code: 0
2016/11/21 18:08:42 [INFO] 0 bytes written for 'stderr'
2016/11/21 18:08:42 packer: 2016/11/21 18:08:42 [INFO] RPC endpoint: Communicator ended with: 0
2016/11/21 18:08:42 [INFO] 208 bytes written for 'stdout'
2016/11/21 18:08:42 [INFO] RPC client: Communicator ended with: 0
2016/11/21 18:08:42 [INFO] RPC endpoint: Communicator ended with: 0
2016/11/21 18:08:42 packer: 2016/11/21 18:08:42 [INFO] RPC client: Communicator ended with: 0
2016/11/21 18:08:42 packer: 2016/11/21 18:08:42 [INFO] 0 bytes written for 'stderr'
2016/11/21 18:08:42 packer: 2016/11/21 18:08:42 [INFO] 208 bytes written for 'stdout'
2016/11/21 18:08:42 ui:     vmware-iso: Zeroing free space on C:\: 0%
    vmware-iso: Zeroing free space on C:\: 0%
2016/11/21 18:08:42 ui: ==> vmware-iso: Gracefully halting virtual machine...
==> vmware-iso: Gracefully halting virtual machine...
2016/11/21 18:08:42 packer: 2016/11/21 18:08:42 Executing shutdown command: shutdown /s /t 10 /f /d p:4:1 /c "Packer Shutdown"
2016/11/21 18:09:45 ui error: ==> vmware-iso: Failed to send shutdown command: unknown error Post http://172.25.243.200:5985/wsman: dial tcp 172.25.243.200:5985: getsockopt: connection timed out
2016/11/21 18:09:45 packer: 2016/11/21 18:09:45 opening new ssh session
==> vmware-iso: Failed to send shutdown command: unknown error Post http://172.25.243.200:5985/wsman: dial tcp 172.25.243.200:5985: getsockopt: connection timed out
2016/11/21 18:09:45 packer: 2016/11/21 18:09:45 starting remote command: vim-cmd vmsvc/power.getstate 43
==> vmware-iso: Destroying virtual machine...
2016/11/21 18:09:46 ui: ==> vmware-iso: Destroying virtual machine...
2016/11/21 18:09:46 packer: 2016/11/21 18:09:46 opening new ssh session
2016/11/21 18:09:46 packer: 2016/11/21 18:09:46 starting remote command: vim-cmd vmsvc/destroy 43

**Comment by trodemaster** [comment]
Just a quick comment.. 

The quoting on this doesn't look correct to me. 
"shutdown_command": "shutdown /s /t 10 /f /d p:4:1 /c "Packer Shutdown""

You might try this..
"shutdown_command": "shutdown /s /t 10 /f /d p:4:1 /c \"Packer Shutdown\""

I'm using the following command successfully on VMware Fusion/Windows Server 2012r2 builds. 
"shutdown_command": "shutdown /s /t 5 /f /d p:4:1"

-Blake

**Comment by moodwrench** [comment]
I think Github's formatting screwed it up.  It's actually:
`"shutdown_command": "shutdown /s /t 10 /f /d p:4:1 /c \"Packer Shutdown\"",`

**Comment by mwhooker** [comment]
Would you please post a minimal json needed to reproduce this?

**Comment by mwhooker** [comment]
going to close until we get a json that reproduces. Will reopen once we have that

**Comment by DaveDingle75** [comment]
@mwhooker I am Using Packer 1.2.2 and am trying to build a Win 7 VM on VMware ESXi.  I am experiencing the 'Gracefully halting  virtual machine' timeout issue when Sysprep is used as the shutdown command.  Would you be interested in taking a look or shall I open a new issue?

**Comment by DaveDingle75** [comment]
Communicator is WinRM, FYI

**Comment by DaveDingle75** [comment]
@mwhooker Looks like my timeout issue was down to the fact that as part of my unattended file I was calling a file which upgrades Powershell to Version 5.  I removed that and the build finished successfully.  Will do a little more testing and report back in case someone else happens to do what I have done.  Thanks.
